python/ray/data/context.py

```diff
@@ -37,6 +38,14 @@
     os.environ.get("RAY_DATASET_PUSH_BASED_SHUFFLE", None)
 )
 
+# Whether to use Polars for tabular dataset groupbys and aggregations.
+try:
```

Comment:
Could we re-organize the code so that there is a common interface for these block operations? Concretely, I'm thinking we can have an `ArrowTransform` interface in `impl/arrow_ops/transform.py`, with implementations in `data/arrow_ops/transform_polars.py` and `data/arrow_ops/transform_simple.py`.

Then, we can have a `get_transform(context) -> ArrowTransformImpl` function and avoid mixing polars code in other files.

Reply:
Hmm not sure if I understand your transform idea exactly, but what about just adding a PolarsArrowBlockAccessor that subclasses ArrowBlockAccessor?

Reply:
That could work too. The main thing is keeping Polars code isolated in a single file.

Reply:
At first glance, I'm more in favor of @ericl's initial composition suggestion rather than subclassing, for a few reason beyond the generic "prefer composition over inheritance" principle:
1. We only want to switch between implementations for a few transformations, not how the Arrow block is represented or how its state is mutated, so isolating that into a "transform" abstraction makes more sense to me.
2. We may want to dynamically switch between transformation implementations based on global config, or at-runtime overrides/dataset-specific automatic configuration, which I think is cleaner to do if dispatching in the method call rather than dispatching at block accessor construction time.

```python
    def sort_and_partition(
        self, boundaries: List[T], key: "SortKeyT", descending: bool, use_polars: bool = None
    ) -> List["Block[T]"]:
        from ray.data.impl.arrow_ops import get_sorter
        context = DatasetContext.get_current()
        # Dispatch.
        sorter = get_sorter(context, use_polars)
        table = sorter(self._table)
```

Reply:
I'm also ok with putting this refactor off, I can do it in the Polars aggregations [PR](https://github.com/ray-project/ray/pull/24282).